File: CRM-Client/src/api/team.ts
import request from '@/utils/request'

export interface RoleSimpleResponse {
  id: number
  name: string
  code: string
}

export interface TeamResponse {
  id: number
  name: string
  code: string
  owner_id: string
  created_at: string
  updated_at?: string
}

export interface TeamCreateRequest {
  name: string
}

export interface TeamJoinRequest {
  code: string
}

export interface TeamMemberResponse {
  id: string
  name: string
  email: string
  avatar_url?: string
  current_team?: boolean
  joined_at: string
  roles: RoleSimpleResponse[]
}

export interface TeamInviteRequest {
  email: string
}

export interface TeamSwitchResponse {
  message: string
  team_id: number
}

export interface UserTeamsListResponse {
  teams: TeamResponse[]
  current_team_id: number
}

export interface AssignRolesRequest {
  role_ids: number[]
}

export const teamApi = {
  createTeam: (data: TeamCreateRequest) => {
    return request.post<TeamResponse>('/v1/teams/', data)
  },

  joinTeam: (data: TeamJoinRequest) => {
    return request.post<TeamResponse>('/v1/teams/join', data)
  },

  getMyTeam: () => {
    return request.get<TeamResponse>('/v1/teams/me')
  },

  getUserTeams: () => {
    return request.get<UserTeamsListResponse>('/v1/teams/user-teams')
  },

  switchTeam: (teamId: number) => {
    return request.post<TeamSwitchResponse>('/v1/teams/switch', { team_id: teamId })
  },

  getTeamDetail: (teamId: number) => {
    return request.get<TeamResponse>(`/v1/teams/${teamId}`)
  },

  getTeamMembers: (teamId: number) => {
    return request.get<TeamMemberResponse[]>(`/v1/teams/${teamId}/members`)
  },

  inviteMember: (teamId: number, data: TeamInviteRequest) => {
    return request.post<{ message: string }>(`/v1/teams/${teamId}/invite`, data)
  },

  addMemberDirect: (teamId: number, userId: number) => {
    return request.post<{ message: string; user_id: number }>(`/v1/teams/${teamId}/members`, { user_id: userId })
  },

  removeMember: (teamId: number, userId: string) => {
    return request.delete<{ message: string }>(`/v1/teams/${teamId}/members/${userId}`)
  },

  regenerateInviteCode: (teamId: number) => {
    return request.post<{ code: string }>(`/v1/teams/${teamId}/regenerate-code`)
  },

  updateTeam: (teamId: number, data: { name?: string }) => {
    return request.put<TeamResponse>(`/v1/teams/${teamId}`, data)
  },

  assignMemberRoles: (teamId: number, userId: string, roleIds: number[]) => {
    return request.post<{ message: string; user_id: number; role_ids: number[] }>(
      `/v1/teams/${teamId}/members/${userId}/roles`,
      { role_ids: roleIds }
    )
  },

  getMemberRoles: (teamId: number, userId: string) => {
    return request.get<RoleSimpleResponse[]>(`/v1/teams/${teamId}/members/${userId}/roles`)
  }
}

File: CRM-Server/app/schemas/team.py
from __future__ import annotations
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class RoleSimpleResponse(BaseModel):
    """角色简要信息"""
    id: int = Field(..., description="角色ID")
    name: str = Field(..., description="角色名称")
    code: str = Field(..., description="角色代码")


class AssignRolesRequest(BaseModel):
    """分配角色请求"""
    role_ids: List[int] = Field(..., description="角色ID列表")


class TeamBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100, description="团队名称")


class TeamCreate(TeamBase):
    pass


class TeamUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100, description="团队名称")


class TeamResponse(TeamBase):
    id: int
    code: str = Field(..., description="邀请码")
    owner_id: int = Field(..., description="创建者ID")
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TeamJoinRequest(BaseModel):
    code: str = Field(..., min_length=1, max_length=20, description="团队邀请码")


class TeamSwitchRequest(BaseModel):
    team_id: int = Field(..., description="要切换的团队ID")


class TeamInviteRequest(BaseModel):
    email: str = Field(..., description="被邀请用户邮箱")


class AddMemberRequest(BaseModel):
    user_id: int = Field(..., description="要添加的用户ID")


class UserTeamResponse(BaseModel):
    id: int
    user_id: int
    team_id: int
    current_team: bool = Field(..., description="是否为当前活跃团队")
    joined_at: datetime

    class Config:
        from_attributes = True


class TeamMemberResponse(BaseModel):
    id: int = Field(..., description="用户ID")
    name: str = Field(..., description="用户姓名")
    email: str = Field(..., description="用户邮箱")
    avatar_url: Optional[str] = Field(None, description="用户头像")
    current_team: bool = Field(..., description="是否为当前活跃团队")
    joined_at: datetime = Field(..., description="加入时间")
    roles: List[RoleSimpleResponse] = Field(default=[], description="用户角色列表")


class TeamWithMembersResponse(TeamResponse):
    members: List[TeamMemberResponse] = []


class UserTeamsListResponse(BaseModel):
    teams: List[TeamResponse]
    current_team_id: Optional[int] = Field(None, description="当前活跃团队ID")


class ResetPasswordRequest(BaseModel):
    """重置成员密码请求"""
    new_password: str = Field(
        ...,
        min_length=6,
        max_length=50,
        description="新密码（6-50位）"
    )